frontend/components/ui/blocknote/blocks/image-block.tsx

```
@@ -0,0 +1,371 @@
+      const [uploading, setUploading] = useState(false);
+      const [isReady, setIsReady] = useState(false);
+      const fileInputRef = useRef<HTMLInputElement>(null);
+      const isMountedRef = useRef(true);
+
+      // Early return if editor or block is not ready
+      if (!props.editor || !props.block) {
+        return null;
+      }
+
+      // Mounting check - delay rendering to ensure editor is fully initialized
+      useEffect(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)